diff --git a/Makefile b/Makefile
index d52c829..6bca153 100644
--- a/Makefile
+++ b/Makefile
@@ -1,21 +1,21 @@
-CC = cc
-CFLAGS = -O2 -Wall -g
-LDFLAGS =
-INSTALL = install
-LIBTOOL = libtool
+CC ?= cc
+CFLAGS ?= -O2 -Wall -g
+
+INSTALL ?= install
+LIBTOOL ?= libtool
 
 name = libchrony
 version = 0.1
 
-libs = -lm
+libs ?= -lm
 
 lib = $(name).la
-lib_version = 0:0:0
+lib_version ?= 0:0:0
 
-prefix = /usr/local
-libdir = $(prefix)/lib
-pkgconfigdir = $(libdir)/pkgconfig
-includedir = $(prefix)/include
+prefix ?= /usr/local
+libdir ?= $(prefix)/lib
+pkgconfigdir ?= $(libdir)/pkgconfig
+includedir ?= $(prefix)/include
 
 objs = $(patsubst %.c,%.o,$(wildcard *.c))
 headers = chrony.h
