From 206f66b6ecc663a2d3da1a50de16ef84924a5fdd Mon Sep 17 00:00:00 2001
From: Erez Geva <ErezGeva2@gmail.com>
Date: Sat, 14 Jun 2025 19:51:13 +0200
Subject: [PATCH] Add clkmgr_proxy and clkmgr.

Signed-off-by: Erez Geva <ErezGeva2@gmail.com>
---
 clknetsim.bash | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/clknetsim.bash b/clknetsim.bash
index 5a55b1f..1d8770a 100644
--- a/clknetsim.bash
+++ b/clknetsim.bash
@@ -88,6 +88,15 @@ start_client() {
 	    args=($opts)
 	    while read line; do args+=("$line"); done <<< "$config"
 	    ;;
+	clkmgr_proxy)
+	    cat > $CLKNETSIM_TMPDIR/conf.$node <<-EOF
+		$config
+		EOF
+	    args=(-f $CLKNETSIM_TMPDIR/conf.$node $opts)
+	    ;;
+	clkmgr)
+	    args=($opts)
+	    ;;
 	*)
 	    echo "unknown client $client"
 	    exit 1
-- 
2.34.1

