From 60cdbc3f8988cde145534a3f057ed4df65050290 Mon Sep 17 00:00:00 2001
From: Erez Geva <ErezGeva2@gmail.com>
Date: Wed, 5 Feb 2025 09:45:56 +0100
Subject: [PATCH] Add futex system call.

Signed-off-by: Erez Geva <ErezGeva2@gmail.com>
---
 client.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/client.c b/client.c
index c7ae1e3..624269f 100644
--- a/client.c
+++ b/client.c
@@ -60,6 +60,7 @@
 #include <linux/pps.h>
 #include <linux/rtc.h>
 #include <linux/sockios.h>
+#include <linux/futex.h>
 #ifdef SO_TIMESTAMPING
 #include <linux/ptp_clock.h>
 #include <linux/net_tstamp.h>
@@ -137,6 +138,7 @@ static int (*_usleep)(useconds_t usec);
 static void (*_srandom)(unsigned int seed);
 static int (*_shmget)(key_t key, size_t size, int shmflg);
 static void *(*_shmat)(int shmid, const void *shmaddr, int shmflg);
+static long (*_syscall)(long number, ...);
 
 static unsigned int node;
 static int initializing = 0;
@@ -290,6 +292,7 @@ static void init_symbols(void) {
 	_srandom = (void (*)(unsigned int seed))dlsym(RTLD_NEXT, "srandom");
 	_shmget = (int (*)(key_t key, size_t size, int shmflg))dlsym(RTLD_NEXT, "shmget");
 	_shmat = (void *(*)(int shmid, const void *shmaddr, int shmflg))dlsym(RTLD_NEXT, "shmat");
+	 _syscall = (long (*)(long number, ...))dlsym(RTLD_NEXT, "syscall");
 
 	initialized_symbols = 1;
 }
@@ -3083,6 +3086,17 @@ long syscall(long number, ...) {
 			}
 			break;
 #endif
+		case __NR_futex:
+			{
+				uint32_t *uaddr = va_arg(ap, uint32_t *);
+				int futex_op = va_arg(ap, int);
+				uint32_t val = va_arg(ap, uint32_t);
+				struct timespec *timeout = va_arg(ap, struct timespec *);
+				uint32_t *uaddr2 = va_arg(ap, uint32_t *);
+				uint32_t val3 = va_arg(ap, uint32_t);
+				r = _syscall(__NR_futex, uaddr, futex_op, val, timeout, uaddr2, val3);
+			}
+			break;
 		default:
 			assert(0);
 	}
-- 
2.39.5

