From abcd052bc6fe7483f4813e2e28bf2708b7a21243 Mon Sep 17 00:00:00 2001
From: Erez Geva <ErezGeva2@gmail.com>
Date: Sun, 9 Feb 2025 01:25:44 +0100
Subject: [PATCH] Add gettid system call.

Signed-off-by: Erez Geva <ErezGeva2@gmail.com>
---
 client.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/client.c b/client.c
index 624269f..ae69bce 100644
--- a/client.c
+++ b/client.c
@@ -3097,6 +3097,9 @@ long syscall(long number, ...) {
 				r = _syscall(__NR_futex, uaddr, futex_op, val, timeout, uaddr2, val3);
 			}
 			break;
+		case __NR_gettid:
+			r = _syscall(__NR_gettid);
+			break;
 		default:
 			assert(0);
 	}
-- 
2.39.5

